test/over_react/util/prop_conversion_test.dart

```diff
@@ -0,0 +1,1067 @@
+        () {
+      group('Map props using (un)jsifyMapProp', () {
+        group('get converted to JS objects', () {
+          group('in the setter, and gets unconverted in getter', () {
+            // This case is a little redundant with the (un)jsifyMapProp tests above, but include it for completeness.
+            test('when set to a Map', () {
+              final builder = TestJs()..buttonProps = {'data-foo': 'bar'};
+
+              final propKey = TestJs.getPropKey((p) => p.buttonProps);
+              expect(builder, {propKey: isA<JsMap>()},
+                  reason:
+                      'test setup: should have converted to a JS object for storage in props map'
+                      ' (we want to ensure this happens before it gets to the ReactComponentFactoryProxy)');
+              expect(builder.buttonProps, isA<Map>(),
+                  reason:
+                      'should have unconverted JsMap to a Map in the typed getter');
+            });
+
+            // This case is a little redundant with the (un)jsifyMapProp tests above, but include it for completeness.
+            test('when null', () {
+              final builder = TestJs();
+
+              expect(builder, {}, reason: 'test setup check');
+              expect(builder.buttonProps, isNull);
+
+              builder.buttonProps = null;
+              final propKey = TestJs.getPropKey((p) => p.buttonProps);
+              expect(builder, {propKey: null});
+              expect(builder.buttonProps, isNull);
+            });
+
+            group('and allows pattern of setting a map prop in a builder', () {
+              test('and then mutating the value read from the builder', () {
+                final builder = TestJs();
+
+                builder.buttonProps = {};
+                expect(builder.buttonProps, {});
+
+                final propKey = TestJs.getPropKey((p) => p.buttonProps);
+                expect(builder, {propKey: isA<JsMap>()},
+                    reason:
+                        'test setup: should have converted to a JS object for storage in props map');
+
+                // For this test, it's important to mutate the the value returned from the `builder.buttonProps` getter,
+                // and not the original Map passed into `builder.buttonProps`.
+                builder.buttonProps['data-foo'] = 'bar';
+                expect(builder.buttonProps, {'data-foo': 'bar'});
+              });
+
+              group('and then reading a nested value that gets unconverted:',
+                  () {
+                group('A nested map prop', () {
+                  test('using the typed props map', () {
+                    final builder = TestJs()
+                      ..buttonProps =
+                          (domProps()..style = {'display': 'block'});
+                    expect(domProps(builder.buttonProps).style,
+                        allOf(isA<Map>(), {'display': 'block'}));
```

Comment:
This isn't quite the same as the original test
```dart
                      ..buttonProps = (Button()..TouchRippleProps = {'foo': 'bar'});
                    expect(Button(builder.buttonProps).TouchRippleProps,
                        allOf(isA<Map>(), {'foo': 'bar'}));
```
since `style` isn't a prop that utilizes the prop conversion utilities like `TouchRippleProps` (well, actually, it kind of does in a special case: https://github.com/Workiva/over_react/pull/698, but it's a different code path than what we want to test here)

Can we replace `style` with another prop that utilizes `jsifyMapProp`/`unjsifyMapProp`?

Reply:
Fixed! Thanks for the help on this